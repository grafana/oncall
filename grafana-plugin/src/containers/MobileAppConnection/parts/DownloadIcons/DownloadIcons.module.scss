.icon {
  width: 25px;
  height: auto;
  margin-right: 12px;
}

.icon-text,
.icon {
  cursor: default;
}

.icon-block {
  display: flex;
  align-items: center;
  min-height: 80px;
  column-gap: 12px;
}

.icon-tag {
  border-radius: 12px;
  font-size: 12px;
  padding: 2px 8px;
  cursor: default;
}
