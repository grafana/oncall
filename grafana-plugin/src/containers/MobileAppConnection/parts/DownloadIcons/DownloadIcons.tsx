import React, { FC } from 'react';

import { VerticalGroup } from '@grafana/ui';
import cn from 'classnames/bind';

import AppleLogoSVG from 'assets/img/apple-logo.svg';
import PlayStoreLogoSVG from 'assets/img/play-store-logo.svg';
import { Block } from 'components/GBlock/Block';
import { Text } from 'components/Text/Text';

import styles from './DownloadIcons.module.scss';

const cx = cn.bind(styles);

export const DownloadIcons: FC = () => (
  <VerticalGroup spacing="lg">
    <Text type="primary" strong>
      Download
    </Text>
    <Text type="primary">The Grafana OnCall app is available on both the App Store and Google Play Store.</Text>
    <VerticalGroup>
      <a
        style={{ width: '100%' }}
        href="https://apps.apple.com/us/app/grafana-oncall-preview/id1669759048"
        target="_blank"
        rel="noreferrer"
      >
        <Block hover fullWidth withBackground bordered className={cx('icon-block')}>
          <img src={AppleLogoSVG} alt="Apple" className={cx('icon')} />
          <Text type="primary" className={cx('icon-text')}>
            iOS
          </Text>
        </Block>
      </a>
      <a
        style={{ width: '100%' }}
        href="https://play.google.com/store/apps/details?id=com.grafana.oncall.prod"
        target="_blank"
        rel="noreferrer"
      >
        <Block hover fullWidth bordered className={cx('icon-block')}>
          <img src={PlayStoreLogoSVG} alt="Play Store" className={cx('icon')} />
          <Text type="primary" className={cx('icon-text')}>
            Android
          </Text>
        </Block>
      </a>
    </VerticalGroup>
  </VerticalGroup>
);
