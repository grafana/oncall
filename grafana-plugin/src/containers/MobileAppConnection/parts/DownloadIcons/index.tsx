import React, { FC } from 'react';

import { VerticalGroup } from '@grafana/ui';
import cn from 'classnames/bind';

import AppleLogoSVG from 'assets/img/brand/apple-logo.svg';
import PlayStoreLogoSVG from 'assets/img/brand/play-store-logo.svg';
import Block from 'components/GBlock/Block';
import Text from 'components/Text/Text';

import styles from './DownloadIcons.module.scss';

const cx = cn.bind(styles);

const DownloadIcons: FC = () => (
  <VerticalGroup spacing="lg">
    <Text type="primary" strong>
      Download
    </Text>
    <Text type="primary">The Grafana IRM app is available on both the App Store and Google Play Store.</Text>
    <VerticalGroup>
      <Block hover fullWidth withBackground bordered className={cx('icon-block')}>
        <img src={AppleLogoSVG} alt="Apple" className={cx('icon')} />
        <Text type="primary" className={cx('icon-text')}>
          iOS
        </Text>
      </Block>
      <a
        style={{ width: '100%' }}
        href="https://play.google.com/store/apps/details?id=com.grafana.oncall.prod"
        target="_blank"
        rel="noreferrer"
      >
        <Block hover fullWidth bordered className={cx('icon-block')}>
          <img src={PlayStoreLogoSVG} alt="Play Store" className={cx('icon')} />
          <Text type="primary" className={cx('icon-text')}>
            Android
          </Text>
        </Block>
      </a>
    </VerticalGroup>
  </VerticalGroup>
);

export default DownloadIcons;
