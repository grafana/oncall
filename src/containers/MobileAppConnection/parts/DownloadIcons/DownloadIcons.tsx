import React, { FC } from 'react';

import { css } from '@emotion/css';
import { Stack, useStyles2 } from '@grafana/ui';
import { StackSize } from 'helpers/consts';

import AppleLogoSVG from 'assets/img/apple-logo.svg';
import PlayStoreLogoSVG from 'assets/img/play-store-logo.svg';
import { Block } from 'components/GBlock/Block';
import { Text } from 'components/Text/Text';

export const DownloadIcons: FC = () => {
  const styles = useStyles2(getStyles);

  return (
    <Stack direction="column" gap={StackSize.lg}>
      <Text type="primary" strong>
        Download
      </Text>
      <Text type="primary">The Grafana OnCall app is available on both the App Store and Google Play Store.</Text>
      <Stack direction="column">
        <a
          style={{ width: '100%' }}
          href="https://apps.apple.com/us/app/grafana-oncall-preview/id1669759048"
          target="_blank"
          rel="noreferrer"
        >
          <Block hover fullWidth withBackground bordered className={styles.iconBlock}>
            <img src={AppleLogoSVG} alt="Apple" className={styles.icon} />
            <Text type="primary" className={styles.iconText}>
              iOS
            </Text>
          </Block>
        </a>
        <a
          style={{ width: '100%' }}
          href="https://play.google.com/store/apps/details?id=com.grafana.oncall.prod"
          target="_blank"
          rel="noreferrer"
        >
          <Block hover fullWidth bordered className={styles.iconBlock}>
            <img src={PlayStoreLogoSVG} alt="Play Store" className={styles.icon} />
            <Text type="primary" className={styles.iconText}>
              Android
            </Text>
          </Block>
        </a>
      </Stack>
    </Stack>
  );
};

const getStyles = () => {
  return {
    icon: css`
      width: 48px;
      height: 48px;
      cursor: default;
    `,

    iconBlock: css`
      display: flex;
      align-items: center;
      min-height: 80px;
      column-gap: 12px;
    `,

    iconText: css`
      margin-left: 16px;
      cursor: default;
    `,

    iconTag: css`
      border-radius: 12px;
      font-size: 12px;
      padding: 2px 8px;
      cursor: default;
    `,
  };
};
