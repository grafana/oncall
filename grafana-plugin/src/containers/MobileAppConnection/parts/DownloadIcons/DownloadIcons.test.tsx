import React from 'react';

import { render } from '@testing-library/react';

import { DownloadIcons } from './DownloadIcons';

describe('DownloadIcons', () => {
  test('it renders properly', () => {
    const component = render(<DownloadIcons />);
    expect(component.container).toMatchSnapshot();
  });
});
