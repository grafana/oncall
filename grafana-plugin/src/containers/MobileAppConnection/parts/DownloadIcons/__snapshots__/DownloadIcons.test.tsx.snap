// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadIcons it renders properly 1`] = `
<div>
  <div
    class="css-gjl87o-vertical-group"
    style="width: 100%; height: 100%;"
  >
    <div
      class="css-12oo3x0-layoutChildrenWrapper"
    >
      <span
        class="root text text--primary text--medium text--strong"
      >
        Download
      </span>
    </div>
    <div
      class="css-12oo3x0-layoutChildrenWrapper"
    >
      <span
        class="root text text--primary text--medium"
      >
        The Grafana OnCall app is available on both the App Store and Google Play Store.
      </span>
    </div>
    <div
      class="css-12oo3x0-layoutChildrenWrapper"
    >
      <div
        class="css-gjl87o-vertical-group"
        style="width: 100%; height: 100%;"
      >
        <div
          class="css-gxt817-layoutChildrenWrapper"
        >
          <a
            href="https://apps.apple.com/us/app/grafana-oncall-preview/id1669759048"
            rel="noreferrer"
            style="width: 100%;"
            target="_blank"
          >
            <div
              class="root root_bordered root--fullWidth root--withBackGround root--hover icon-block"
            >
              <img
                alt="Apple"
                class="icon"
                src="[object Object]"
              />
              <span
                class="root text text--primary text--medium icon-text"
              >
                iOS
              </span>
            </div>
          </a>
        </div>
        <div
          class="css-gxt817-layoutChildrenWrapper"
        >
          <a
            href="https://play.google.com/store/apps/details?id=com.grafana.oncall.prod"
            rel="noreferrer"
            style="width: 100%;"
            target="_blank"
          >
            <div
              class="root root_bordered root--fullWidth root--hover icon-block"
            >
              <img
                alt="Play Store"
                class="icon"
                src="[object Object]"
              />
              <span
                class="root text text--primary text--medium icon-text"
              >
                Android
              </span>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
